### PR TITLE
Fixed a bug where daemonsets weren't correctly included in binpacking calculations

### PR DESCRIPTION
**Pull request by ellistarn**
**1. Issue, if available:**

**2. Description of changes:**

**3. Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: *link to issue*
- [ ] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/controllers/provisioning/scheduling/scheduler.go [hunk 1]
```diff
@@ -147,17 +147,15 @@ func (s *Scheduler) getSchedules(ctx context.Context, constraints *v1alpha5.Cons
...
 		return nil, fmt.Errorf("listing daemonsets, %w", err)
 	}
-
-	// 2. filter DaemonSets to include those that will schedule on this node
+	// Include daemonsets that will schedule on this node
 	pods := []*v1.Pod{}
 	for _, daemonSet := range daemonSetList.Items {
 		pod := &v1.Pod{Spec: daemonSet.Spec.Template.Spec}
-		if constraints.Supports(pod) != nil {
+		if constraints.Supports(pod) == nil {
```
Comment (JacobGabrielson) [inline pkg/controllers/provisioning/scheduling/scheduler.go:158]: I think part of the issue here is that the name `Supports` (which I think reads most naturally as "supports?") suggests a boolean (true/false)? Is there some reason this has to return a nullable value?

File: pkg/controllers/provisioning/suite_test.go [hunk 2]
```diff
@@ -134,33 +133,81 @@ var _ = Describe("Provisioning", func() {
...
+					test.PodOptions{
+						Tolerations:          []v1.Toleration{{Operator: v1.TolerationOperator(v1.NodeSelectorOpExists)}},
+						ResourceRequirements: v1.ResourceRequirements{Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("1"), v1.ResourceMemory: resource.MustParse("1Gi")}},
+					},
+				))[0]
+				node := ExpectScheduled(ctx, env.Client, pod)
+				Expect(*node.Status.Allocatable.Cpu()).To(Equal(resource.MustParse("2")))
+				Expect(*node.Status.Allocatable.Memory()).To(Equal(resource.MustParse("2Gi")))
+			})
+			It("should ignore daemonsets an invalid selector", func() {
```
Comment (JacobGabrielson) [inline pkg/controllers/provisioning/suite_test.go:178]: missing word before `an`

File: pkg/controllers/provisioning/suite_test.go [hunk 3]
```diff
@@ -134,33 +133,81 @@ var _ = Describe("Provisioning", func() {
...
+					test.PodOptions{
+						ResourceRequirements: v1.ResourceRequirements{Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("1"), v1.ResourceMemory: resource.MustParse("1Gi")}},
+					},
+				))[0]
 				node := ExpectScheduled(ctx, env.Client, pod)
 				Expect(*node.Status.Allocatable.Cpu()).To(Equal(resource.MustParse("4")))
 				Expect(*node.Status.Allocatable.Memory()).To(Equal(resource.MustParse("4Gi")))
-			}
+			})
+			It("should not schedule if overhead is too large", func() {
```
Comment (JacobGabrielson) [inline pkg/controllers/provisioning/suite_test.go:152]: Is the reason why this test works that none of the instance types in the fake EC2 implementation have 10Gi of memory? I'd probably at least add a comment to that effect, especially considering this is the main test that tests out that functionality. Also, why not make the daemonset require way more memory and 10Gi just to make it more clear?

Comment (ellistarn) [reply pkg/controllers/provisioning/suite_test.go:152]: Good call.

File: pkg/controllers/provisioning/scheduling/scheduler.go (hunk 1 above)
Comment (ellistarn) [reply pkg/controllers/provisioning/scheduling/scheduler.go]: This is going to change a bit soon. I want to cover this in a refactor, but for now, I've renamed to ValidatePod

**Review by JacobGabrielson — APPROVED** [review]
🚢

**Pull request merged**